spock-specs/src/test/groovy/org/spockframework/smoke/extension/PendingFeatureIfExtensionSpec.groovy

```diff
@@ -81,11 +81,11 @@ def bar() {
 
     then:
     notThrown(AssertionError)
-    result.testsStartedCount == 1
+    result.testsStartedCount == 2
```

Comment:
This is from and due to #1100

Reply:
I understand Leonard that he would like to have the behavior with rolled up feature tested - which was done originally (probably "accidentally" as it was the default behavior).

However, as #1100 changes the way how iterations and a container are counted (which seems reasonable as in 2.0-M2, `@PendingFeature` doesn't work properly with parameterized tests - all the failing tests are marked as successful, so it might be fixed) we may want to have also the new tests for unrolled behavior (now enabled by default).

Reply:
Pending feature fix is a separate PR based on this one.
Actually that is what I originally wanted to fix and could not properly because of the half finished default unroll. :-D
You can find it at #1103, there also all combinations should be tested I think.

Reply:
> You can find it at #1103, there also all combinations should be tested I think.

Ok, I have found those tests for [`@PendingFeature`](https://github.com/spockframework/spock/pull/1103/commits/08b34095f165aca52a61ca3b26e116d041de6434#diff-305ac92ee940c2faee974bd0dac762e9R155-R205
). I wonder if there is a need to have something similar also for `@IgnoreIf/@Required` (if it share the same codebase)?


> Actually that is what I originally wanted to fix and could not properly because of the half finished default unroll. :-D

I couldn't resist to paste this "meme" :-)
https://imgur.com/gallery/rQIb4Vw

Reply:
Yeah, good ol' dad of Malcom :-D

I don't know whether additional tests for `@IgnoreIf` and `@Required` are necessary, but if so, I think it is out of scope of this PR or #1103.
If there were only tests for one state of unrolled / uprolled, then this can surely be improved, but probably as a separate change imho.

Reply:
@leonard84 WDYT?